Week 1 and 2

chanyijuan opened the pull request:
Added basic Level-1 to Level-7 implementations.
Level-5: Error handling halfway done.

Commit:
Add Gradle support

chanyijuan committed:
Level-1: Implemented with while-loop and Scanner class

chanyijuan committed:
Level-2: Implemented nested condition in while-loop

chanyijuan committed:
Level-3: Implemented A-Class for Commands

chanyijuan committed:
Level-4: ToDos, Deadlines, Events. Implemented with A-Inheritance.

chanyijuan committed:
A_TextUiTesting: Added automatic testing, corrected line spacing.

chanyijuan committed:
Level-6: Changed HashMap<Integer, Task> to Arraylist<Task> to support delete function. Updated autotesting to validate delete function.

chanyijuan committed:
Level-7:
- Added file output stream to save updated tasks to tasks.json (Duke.java)
- implemented Serializable class (Task.java)

raynerljm commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,112 @@
 public class Duke {
+    static String indent = "     ";
+    static String line = "    ____________________________________________________________";
+    private static final String FILEPATH = ""
```
Good job making the constant variable in screaming snake case.

Justinnje replied:
I agree. Common Naming mistake. Good Job!

raynerljm commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,112 @@
-        System.out.println("Hello from\n" + logo);
+        // Greeting
+        System.out.println(line);
+        System.out.println(indent+"Hello! I'm Duke\n"+
```
You should leave spaces on the left and right of operators

raynerljm commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,112 @@
+        // Greeting
+        System.out.println(line);
+        System.out.println(indent+"Hello! I'm Duke\n"+
+                indent+
```
You should let the '+' be the start of the newline

raynerljm commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,112 @@
+        System.out.println();
+
+        // Data Structures
+        ArrayList<Task> commands = new ArrayList<>();
```
Good job naming the arraylist with a plural variable name

raynerljm commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,112 @@
+            if (cmd.equals("list")) {
+
+                System.out.println(indent+"Here are the tasks in your list:");
+                for (int i=0; i<commands.size(); i++) {
```
Note the spaces around operators.

raynerljm commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,15 @@
+import java.lang.*;
```
Don't use wild card imports

raynerljm commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,26 @@
+import java.lang.*;
+import java.util.*;
```
Same thing about wildcard imports

raynerljm reviewed (commented):
Overall good job. However, do note some of the coding standards. It would be good to setup checkstyles so it automatically catches the discrepancies for you.

Justinnje commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,112 @@
+    public static void updateTask(Task task) {
+        try {
+            FileOutputStream f = new FileOutputStream(new File("tasks.json"));
+            ObjectOutputStream o = new ObjectOutputStream(f);
```
Maybe you could use more descriptive variable names rather than f and o. Similarly for fi and oi at line 21 and 22.

Justinnje commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,15 @@
+        super(rank, description.split(" /by ")[0]);
+        this.deadline = description.split(" /by ")[1];
+    }
+
+    public String toString() {
+        return "[D]" + super.toString() + " (by: " + this.deadline + ")";
```
I think there should be an @Override tag above your toString() methods.

Justinnje commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,112 @@
+                    System.out.println(indent+(i+1)+"."+commands.get(i));
+                }
+
+            } else if (cmd.equals("mark") || cmd.equals("unmark")) {
+
+                int index = Integer.parseInt(st.nextToken());
+                commands.get(index-1).setDone(cmd.equals("mark"));
+                if (cmd.equals("mark")) {
+                    System.out.println(indent + "Nice! I've marked this task as done:");
+                } else {
+                    System.out.println(indent + "OK, I've marked this task as not done yet:");
+                }
+                System.out.println(indent + "  " + commands.get(index-1));
+
```
Maybe remove the Overarching else if check, as there is a nested check? Makes code seem messy.

Justinnje reviewed (commented):
Overall Good Job! Just some small details to take note off. Also, couldn't comment on this below, but you may want to update your Pull Request Name. Should be [Name] iP.

chanyijuan committed:
Level-7: Save
* load serialized tasks to tasks.txt before exiting Duke (TaskList.java)
* refactored code to shorter methods (Duke.java)

chanyijuan committed:
Level-8 Date:
* takes in deadline date in ISO_LOCAL_DATE format and prints in d-MMM-yyyy format

chanyijuan committed:
A-More-OOP:
* Added Parser class to parse through commands (Parser.java)
* Added abstract Command class to initialise Task objects and perform commands based on Command::execute
	* Added ListCommand, ExitCommand, DeleteCommand, MarkCommand, UnmarkCommand, TodoCommand, DeadlineCommand, EventCommand
* Added Ui class to read user input and produce output (confirmation and error messages)

chanyijuan committed:
A-Packages:
* created duke package
* organised classes into tasks, parser, commands, exception, ui, and storage packages

chanyijuan committed:
A-JUnit:
* added JUnit test classes to test:
	* Todo object instantiation
	* DeadlineCommand Deadline object instantiation

chanyijuan committed:
A-JavaDocs:
* added JavaDocs to all files
* TODO: @docRoot

chanyijuan committed:
Level-9:
* added FindCommand class to execute the task finding
* added a COMMAND_FIND constant to Parser to parse out a FindCommand object
* added a KeywordFinder interface to implement a method to find tasks with the keyword

chanyijuan committed:
A-CodingStandard:
* adjusted boolean variable names to sound like booleans
* adjusted long method arguments to fit max length
* adjusted whitespacing between math operators
* renamed class fields to be more consistent
* removed indent for switch statement
* removed wildcard import in Storage class

chanyijuan committed:
no message

chanyijuan committed:
Merge branch 'branch-Level-9'

* branch-Level-9:
  Level-9: * added FindCommand class to execute the task finding * added a COMMAND_FIND constant to Parser to parse out a FindCommand object * added a KeywordFinder interface to implement a method to find tasks with the keyword

# Conflicts:
#	src/main/java/duke/parser/Parser.java

chanyijuan committed:
Merge branch 'add-gradle-support'

* add-gradle-support:
  Add Gradle support

chanyijuan committed:
Set theme jekyll-theme-hacker

chanyijuan committed:
Set theme jekyll-theme-hacker

chanyijuan committed:
A-Checkstyle:
* corrected code quality according to Gradle Checkstyle

chanyijuan committed:
A-UserGuide:
* Edited program outputs to be more consistent
* Edited User Guide
